nns/nns_contract.go

```diff
@@ -452,11 +452,8 @@ func Resolve(name string, typ RecordType) []string {
 
 // GetAllRecords returns an Iterator with RecordState items for the given name.
 func GetAllRecords(name string) iterator.Iterator {
-	tokenID := []byte(tokenIDFromName(name))
 	ctx := storage.GetReadOnlyContext()
-	_ = getNameState(ctx, tokenID) // ensure not expired
-	recordsKey := getRecordsKey(tokenID, name)
-	return storage.Find(ctx, recordsKey, storage.ValuesOnly|storage.DeserializeValues)
+	return getAllRecords(ctx, name)
```

Comment:
If we have a `maxRecordID` and a limited set of types, maybe we can avoid having `iterator.Iterator`? Although we still should have some reasonable number of items in the end. Just an idea.

Reply:
Maybe we firstly need to discuss `maxRecordID` value? I've added this constraint because of the record ID type: it was `byte`, but all int-like go types are `BiGInteger` in the VM, so technically, the previous implementation didn't restrict the maximum number of records of the same type. I suppose, it's not the initial intention, thus let's decide on `maxRecordID`.

And if we're to keep `MaxRecordID` set to be 255, then I vote for converting the result of `getAllecords` to `[]stirng`.

Reply:
DNS doesn't have this limit. C# NNS doesn't have it too. But I see what you're talking about now, `id byte` overflow can be a problem in some cases. It can be fixed at the same time to have unlimited number of records of any type. Practically I'd rather have some limit (16-64), at the moment C# contract has a limit of 1 IIRC and it's kinda OK.

Reply:
@fyrchik, what do you think about it?

Reply:
I think `[]string` is easier to work with. In our scenarios we have at most 2 records per type, so 255 is more than enough.

Reply:
Getting back to this discussion, the profit of `Iterator` returned value here is that it returns not only record name, but the complete record information which includes record's name,  type, ID and data. Currently it's the only method allowing to retrieve the whole `RecordState` from the contract, so I've changed my mind and suggest to keep `Iterator`. @roman-khimov, somehow I've missed it previously.